routers/web/repo/view.go

```diff
@@ -489,9 +489,17 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		}
 
 		rd := charset.ToUTF8WithFallbackReader(io.MultiReader(bytes.NewReader(buf), dataRc))
-		readmeExist := markup.IsReadmeFile(blob.Name())
+
+		shouldRenderMarkdown := ctx.FormString("display") != "source"
+		readmeExist := markup.IsReadmeFile(blob.Name()) && shouldRenderMarkdown
```

Comment:
Seems not correct. Why `readmeExist` is affected by `shouldRenderMarkdown`?

Reply:
See a few lines down, otherwise it will first not render "correctly" with line numbers.

Reply:
However, `ReadmeExist` controls template logic, I do not know how could it correct if the meaning of `ReadmeExist` changes.

https://github.com/go-gitea/gitea/blob/d906858847e0d345bc02f04b4037dfa80650889e/templates/repo/view_list.tmpl#L95-L97

Shouldn't it be something like `else if readmeExist && shouldRenderMarkdown {` below?

Reply:
Moved the logic to the if statement. Indeed better.